code/src/components/app-select/AppSelect.tsx

```diff
@@ -9,33 +9,41 @@ import cn from "@/utils/cn/cn";
 
 import "@/components/app-select/AppSelect.scss";
 
-const AppSelect = ({
-  label,
-  labelId,
-  color = "contained",
-  className,
-  ...props
-}: AppSelectProps) => {
-  return (
-    <FormControl>
-      {label && (
-        <InputLabel
-          className={cn("spa-select__label", `spa-select__label--${color}`)}
-          id={`spa-select-label-${labelId}`}
-          data-testid="spa-select-label"
-        >
-          <AppTypography translationKey={label} />
-        </InputLabel>
-      )}
-      <Select
-        className={cn("spa-select", `spa-select--${color}`, className)}
-        labelId={`spa-select-label-${labelId}`}
-        id={`spa-select-${labelId}`}
-        label={label}
-        {...props}
-      />
-    </FormControl>
-  );
-};
+const AppSelect = forwardRef(
+  (
+    {
+      label,
+      labelId,
+      color = "contained",
+      className,
+      ...props
+    }: AppSelectProps,
+    ref
+  ) => {
+    return (
+      <FormControl>
+        {label && (
+          <InputLabel
+            className={cn("spa-select__label", `spa-select__label--${color}`)}
+            id={`spa-select-label-${labelId}`}
+            data-testid="spa-select-label"
+          >
+            <AppTypography translationKey={label} />
+          </InputLabel>
+        )}
+        <Select
+          ref={ref}
+          className={cn("spa-select", `spa-select--${color}`, className)}
+          labelId={`spa-select-label-${labelId}`}
+          id={`spa-select-${labelId}`}
+          label={label}
+          {...props}
+        />
+      </FormControl>
+    );
```

Comment:
Why do we pass `labelId` as prop and not using `useId` hook here? It seems redundant because we will have to specify this id over and over again in the parent component and if we won't use `useId` hook or specify unique value, we will eventually run into issues with duplicate ids on different selects

Reply:
used